src/steps/os/unix.rs

```diff
@@ -1,21 +1,42 @@
-use crate::error::Error;
+use crate::error::{Error, ErrorKind::*};
 use crate::executor::{CommandExt, RunType};
 use crate::terminal::print_separator;
 use crate::utils::{require, which};
 use directories::BaseDirs;
+use std::env;
 use std::path::Path;
 use std::process::Command;
 
+fn zplug_exists(base_dirs: &BaseDirs) -> bool {
+    let home_exists = match env::var("ZPLUG_HOME") {
+        Ok(ref zplug_home) => Path::new(zplug_home).exists(),
+        Err(_) => false,
+    };
+    let dotdir_exists = base_dirs.home_dir().join(".zplug").exists();
+    home_exists || dotdir_exists
+}
+
+fn get_zshrc(base_dirs: &BaseDirs) -> Result<String, ()> {
+    let zshrc = match env::var("ZDOTDIR") {
```

Comment:
You can use [ok_or_else](https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or_else)